detectors/aws/ecs/ecs.go

```diff
@@ -97,12 +100,12 @@ func (detector *resourceDetector) Detect(ctx context.Context) (*resource.Resourc
 	}
 
 	if len(metadataURIV4) > 0 {
-		containerMetadata, err := ecsmetadata.GetContainerV4(ctx, &http.Client{})
+		containerMetadata, err := detector.utils.getContainerMetadataV4(ctx)
 		if err != nil {
 			return empty, err
 		}
 
-		taskMetadata, err := ecsmetadata.GetTaskV4(ctx, &http.Client{})
+		taskMetadata, err := detector.utils.getTaskMetadataV4(ctx)
```

Comment:
It seems superfluous to add a method that just makes the same direct function call. Why are `getTaskMetadataV4` and `getContainerMetadataV4` needed?

Reply:
Because mocking. Without those one liners, I could not achieve good functional coverage (as opposed to the line coverage the tooling complains about).

Reply:
I got the gist from the test infrastructure that the mock facilities need them to be methods

Reply:
Gotcha :+1: